.github/actions/publish-nuget/action.yml

```diff
@@ -114,7 +118,15 @@ runs:
     # This also implicitly restores NuGet dependencies.
     - name: Add Source Link package
       shell: pwsh
-      run: Add-SourceLinkPackage
+      run: |
+        if ('${{ inputs.add-source-link-package }}' -eq 'true')
+        {
+          Add-SourceLinkPackage
```

Comment:
How does this fail with Cloudsmith?

Reply:
Like this: https://github.com/Lombiq/Helpful-Libraries/actions/runs/9643961750/job/26595027509#step:4:288

> error: Package 'Microsoft.SourceLink.GitHub' is incompatible with 'all' frameworks in project '/home/runner/work/Helpful-Libraries/Helpful-Libraries/Lombiq.HelpfulLibraries.LinqToDb/Lombiq.HelpfulLibraries.LinqToDb.csproj'.

(All the other "Unable to resolve" errors in the log are irrelevant and they also went away after I turned off `Microsoft.SourceLink.GitHub`.)

Reply:
So it tries to restore all packages, as we observed earlier (hence the code in the script) but it only uses nuget.org and thus fails. Isn't this fixed by removing `--source` [here](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/publish-nuget/Add-SourceLinkPackage.ps1#L48)? Only if this still works if there's no NuGet.config.

Reply:
Perhaps. Do we know why was `--source 'https://api.nuget.org/v3/index.json'` necessary in the first place?

Reply:
> Only if this still works if there's no NuGet.config.

Yes, that works without `--source`:
![image](https://github.com/user-attachments/assets/73706700-bc4d-495b-af2d-60b1b6b2d880)

Reply:
Great, then we can just use that.

I think when writing that my understanding was that `--source` will use that source to restore SourceLink itself. If it works without it, then nothing else to do.